src/web/app/user-join-page.component.ts

```diff
@@ -43,37 +47,70 @@ export class UserJoinPageComponent implements OnInit {
-          this.navigationService.navigateByURL(this.router, `/web/${this.entityType}/home`);
-        } else {
-          this.isLoading = false;
-        }
-      }, (resp: ErrorMessageOutput) => {
-        if (resp.status === 403) {
-          this.isLoading = false;
-          const nextUrl: string = `${window.location.pathname}${window.location.search.replace(/&/g, '%26')}`;
-          this.authService.getAuthUser(undefined, nextUrl).subscribe((auth: AuthInfo) => {
-            if (!auth.user) {
-              window.location.href = `${this.backendUrl}${auth.studentLoginUrl}`;
-            }
-          });
-        } else {
-          const modalRef: any = this.ngbModal.open(ErrorReportComponent);
-          modalRef.componentInstance.requestId = resp.error.requestId;
-          modalRef.componentInstance.errorMessage = resp.error.message;
-        }
-      });
+      if (this.isCreatingAccount) {
+        this.setupForCreateAccount();
+      } else {
+        this.setupForJoinCourse();
+      }
+    });
+  }
+
+  /**
+   * Setup page for instructor to join course.
+   */
+  setupForJoinCourse(): void {
+    this.courseService.getJoinCourseStatus(this.key, this.entityType).subscribe((resp: JoinStatus) => {
+      this.hasJoined = resp.hasJoined;
+      this.userId = resp.userId || '';
+      if (this.hasJoined && this.userId) {
+        // The regkey has been used and there is a logged in user.
+        // Simply redirect the user to their home page, regardless of whether the regkey matches or not.
+        this.navigationService.navigateByURL(this.router, `/web/${this.entityType}/home`);
+      } else {
+        this.isLoading = false;
+      }
+    }, (resp: ErrorMessageOutput) => {
+      if (resp.status === 403) {
+        this.isLoading = false;
+        const nextUrl: string = `${window.location.pathname}${window.location.search.replace(/&/g, '%26')}`;
+        this.authService.getAuthUser(undefined, nextUrl).subscribe((auth: AuthInfo) => {
+          if (!auth.user) {
+            window.location.href = `${this.backendUrl}${auth.studentLoginUrl}`;
+          }
+        });
+      } else {
+        const modalRef: any = this.ngbModal.open(ErrorReportComponent);
+        modalRef.componentInstance.requestId = resp.error.requestId;
+        modalRef.componentInstance.errorMessage = resp.error.message;
+      }
+    });
+  }
+
+  /**
+   * Setup page for instructor to create account.
+   */
+  setupForCreateAccount(): void {
+    const nextUrl: string = `${window.location.pathname}${window.location.search.replace(/&/g, '%26')}`;
+    this.authService.getAuthUser(undefined, nextUrl).subscribe((resp: AuthInfo) => {
+      this.userId = resp.user?.id || '';
+
+      if (!resp.user) {
+        window.location.href = `${this.backendUrl}${resp.instructorLoginUrl}`;
+      }
+
+      if (resp.user?.isInstructor) {
+        // User already has instructor account
```

Comment:
This is not sufficient. Create account is an action that transforms an account request to a set of sample data. Just because a user is already an existing instructor in the system, does not mean that the existing instructor corresponds to the requested account creation.